lib/ZumoOta/src/Upload.cpp

```diff
@@ -59,14 +60,93 @@
  * Public Methods
  *****************************************************************************/
 
-Upload::Upload() {
-   
+Upload::Upload()
+{   
+}
+
+Upload::~Upload()
+{    
+}
+
+void Upload::handleUploadButtonPress(AsyncWebServerRequest *request)
+{
+    request->send(200, "text/plain", "Upload Button gedrueckt");
+ 
 }
 
-Upload::~Upload() {
-    
+void Upload::handleFileUpload(AsyncWebServerRequest *request, const String& filename, size_t index, uint8_t *data, size_t len, bool final)
+{
+    static File file;
+    String updatedFilename = filename;
+
+    if (!filename.startsWith("/"))
+    {
+        updatedFilename = "/" + filename;
+    }
+
+    if (!index)
+    {
+        // Dies ist der erste Datenblock, Datei öffnen oder erstellen
+        file = LittleFS.open(updatedFilename, "w");
+
+        if (!file)
+        {
+            LOG_DEBUG("Failed to open file for writing");
+            file.close();
+            return;
+        }
+    }
+    if (file)
+    {
+      // Datei im Request-Objekt speichern
+      request->_tempFile = file;
+      LOG_DEBUG("Upload Start: " + String(updatedFilename));
+    }
+    else
+    {
+      return;
+    }
+    if(len)
+    {
+        // Daten in die Datei schreiben
+        request->_tempFile.write(data, len);
+    }
+
+    // Wenn dies der letzte Datenblock ist, Datei schließen
+    if (final)
+    {
+        file.close();
+        request->redirect("/filelist");
+    }
+
+    // Überprüfe, ob die Datei im Dateisystem liegt
+    if (isFileUploaded(request, updatedFilename))
+    {
+        LOG_DEBUG("Die hochgeladene Datei liegt im Dateisystem.");
+    }
+    else
+    {
+        LOG_DEBUG("Die hochgeladene Datei liegt nicht im Dateisystem.");

```

Comment:
english

Reply:
It will be taken care of by the next commit.